packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import {
+  Command,
+  SfdxCommandBuilder,
+  SfdxProjectJsonParser
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { PostconditionChecker } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import {
+  CancelResponse,
+  ContinueResponse
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types/index';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import {
+  FilePathGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+export class ForceSourceRetrieveSourcePathExecutor extends SfdxCommandletExecutor<
+  string
+> {
+  public build(sourcePath: string): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_source_retrieve_text'))
+      .withArg('force:source:retrieve')
+      .withFlag('--sourcepath', sourcePath)
+      .withLogName('force_source_retrieve_with_sourcepath')
+      .build();
+  }
+}
+
+export class SourcePathChecker implements PostconditionChecker<string> {
+  public async check(
+    inputs: ContinueResponse<string> | CancelResponse
+  ): Promise<ContinueResponse<string> | CancelResponse> {
+    if (inputs.type === 'CONTINUE') {
+      const sourcePath = inputs.data;
+      const sfdxProjectPath = vscode.workspace!.workspaceFolders![0].uri.fsPath;
+      const sfdxProjectJsonParser = new SfdxProjectJsonParser();
+      try {
+        const packageDirectoryPaths = await sfdxProjectJsonParser.getPackageDirectoryPaths(
+          sfdxProjectPath
+        );
+        const fullPackagePaths = packageDirectoryPaths.map(
+          packageDirectoryPath =>
+            path.join(sfdxProjectPath, packageDirectoryPath)
+        );
+
+        let sourcePathIsInPackageDirectory = false;
+        for (const packagePath of fullPackagePaths) {
+          if (sourcePath.startsWith(packagePath)) {
```

Comment:
Wouldn't `packagePath` be something like `force-app` and `sourcePath` be something like `Users/myuser/repo_name/force-app` ?

Reply:
Disregard this, I've checked and I was wrong 👍